Update file sync planning with common sync scenarios and solutions

jithubhaijabs opened the pull request:
These are scenarios we see customers struggling with until they get a good understanding of File sync solution, its limits, supported config etc. On-premise file server solutions usually have complex, not very well-managed file servers and structure. Also many a times customer doesn't think about the Azure files or Storage account limits and want consolidated single file share. This table address most common scenarios we see with file servers and sync requirements.
Note: this work item has been discussed with Mike already.

jithubhaijabs committed:
Update file sync planning with common sync scenarios and solutions

These are scenarios we see customers struggling with until they get a good understanding of File sync solution, its limits, supported config etc. On-premise file server solutions usually have complex, not very well-managed file servers and structure. Also many a times customer doesn't think about the Azure files or Storage account limits and want consolidated single file share. This table address most common scenarios we see with file servers and sync requirements.
Note: this work item has been discussed with Mike already.

jithubhaijabs committed:
Merge branch 'MicrosoftDocs:main' into master

jithubhaijabs committed:
Update storage-files-migration-namespace-mapping.md

Fixed url links in the new section added "Common File Sync Scenarios and Considerations"

jithubhaijabs committed:
Update storage-files-migration-namespace-mapping.md

jithubhaijabs committed:
Merge branch 'MicrosoftDocs:main' into master

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+
+| # | Sync Scenarios | Supported | Considerations (or limitations) | Solution (or workaround) | 
+|---|---|:---:|---|---|
+| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume wil ...
```
```suggestion
| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-premises. Configure cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target one root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure file shares (same or different storage account based on performance requirements) |
```

khdownie replied:
```suggestion
| 1 | File server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-prem. Config cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target 1 root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure File Share (same or different storage account based on perf requirements) |
```

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+| # | Sync Scenarios | Supported | Considerations (or limitations) | Solution (or workaround) | 
+|---|---|:---:|---|---|
+| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume wil ...
+| 2 | File Server with single volume and multiple shares to same target Azure File share (Consolidation) | Yes* | Cannot have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root* of the volume holding multiple shares or top-level folders. Refer [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more info. |
```
```suggestion
| 2 | File server with single volume and multiple shares to same target Azure file share (consolidation) | Yes | Can't have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root of the volume holding multiple shares or top-level folders. Refer to [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more information. |
```

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+|---|---|:---:|---|---|
+| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume wil ...
+| 2 | File Server with single volume and multiple shares to same target Azure File share (Consolidation) | Yes* | Cannot have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root* of the volume holding multiple shares or top-level folders. Refer [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more info. |
+| 3 | File Server with multiple shares and/or volumes to multiple Azure file shares under single Storage Account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep no. of items per sync group within 100 million ...
```
```suggestion
| 3 | File server with multiple shares and/or volumes to multiple Azure file shares under single storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep number of items per sync group within 100 million items (files and folders) per share. Ideally it's best to stay below 20 or 30 million per share. | 1) Use multiple sync groups (number of sync groups = number of Azure file shares to sync to).<br/>  2) Only 30 shares can be synced in this scenario at a time. If you have more than 30 shares on that file server, use [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) to reduce the number of root or top-level folders at source.<br/> 3) Use additional File Sync servers on-premises and split/move data to these servers to work around limitations on the source Windows server. |
```

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume wil ...
+| 2 | File Server with single volume and multiple shares to same target Azure File share (Consolidation) | Yes* | Cannot have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root* of the volume holding multiple shares or top-level folders. Refer [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more info. |
+| 3 | File Server with multiple shares and/or volumes to multiple Azure file shares under single Storage Account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep no. of items per sync group within 100 million ...
+| 4 | File Server with multiple shares and/or volumes to multiple Azure file shares under different Storage Account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares (same or different Storage account).<br/><br/> Keep no. of items per sync group within 100 million items (files and folders) per share. Ideally best if it's below 20 or 30 m ...
```
```suggestion
| 4 | File server with multiple shares and/or volumes to multiple Azure file shares under different storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares (same or different storage account).<br/><br/> Keep number of items per sync group within 100 million items (files and folders) per share. Ideally it's best to stay below 20 or 30 million per share. | Same approach as above |
```

jithubhaijabs committed:
Clarify DFS-N and File sync agent statement

Changes based on the issue raised - https://github.com/MicrosoftDocs/azure-docs/issues/97683

jithubhaijabs committed:
Merge pull request #1 from jithubhaijabs/jithubhaijabs-patch-1

Clarify DFS-N and File sync agent statement

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Merge branch 'MicrosoftDocs:main' into master

khdownie commented on articles/storage/file-sync/file-sync-planning.md:
```diff
@@ -242,7 +242,7 @@ Azure File Sync does not support Data Deduplication and cloud tiering on the sam
 Azure File Sync supports interop with DFS Namespaces (DFS-N) and DFS Replication (DFS-R).
 
-**DFS Namespaces (DFS-N)**: Azure File Sync is fully supported on DFS-N servers. You can install the Azure File Sync agent on one or more DFS-N members to sync data between the server endpoints and the cloud endpoint. For more information, see [DFS Namespaces overview](/windows-server/storage/dfs-namespaces/dfs-overview).
+**DFS Namespaces (DFS-N)**: Azure File Sync is fully supported with DFS-N implementation. You can install the Azure File Sync agent on one or more file servers to sync data between the server endpoints and the cloud endpoint and then use DFS-N to provide namespace service. For more information, see [DFS Namespaces overview](/windows-server/storage/dfs-namespaces/dfs-overview) and [DFS Namespaces  ...
```
```suggestion
**DFS Namespaces (DFS-N)**: Azure File Sync is fully supported with DFS-N implementation. You can install the Azure File Sync agent on one or more file servers to sync data between the server endpoints and the cloud endpoint, and then use DFS-N to provide namespace service. For more information, see [DFS Namespaces overview](/windows-server/storage/dfs-namespaces/dfs-overview) and [DFS Namespaces with Azure Files](../files/files-manage-namespaces.md).
```

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+
+| # | Sync Scenarios | Supported | Considerations (or limitations) | Solution (or workaround) | 
+|---|---|:---:|---|---|
+| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volu ...
```
```suggestion
| 1 | File server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-premises. Configure cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target one root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync, and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure file shares (same or different storage account based on performance requirements) |
```

khdownie commented on includes/storage-files-migration-namespace-mapping.md:
```diff
@@ -60,6 +60,16 @@ It's a best practice to keep the number of items per sync scope low. That's an i
+| 2 | File server with single volume and multiple shares to same target Azure file share (consolidation) | Yes | Can't have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root of the volume holding multiple shares or top-level folders. Refer to [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more informa ...
+| 3 | File server with multiple shares and/or volumes to multiple Azure file shares under single storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep number of items per sync group within 100 mill ...
+| 4 | File server with multiple shares and/or volumes to multiple Azure file shares under different storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares (same or different storage account).<br/><br/> Keep number of items per sync group within 100 million items (files and folders) per share. Ideally it's best to stay below 20 ...
+| 5 | Multiple File Servers with single (root volume or share) to same target Azure File share (Consolidation) | No | A sync group cannot use cloud endpoint (Azure file share) already configured in another sync group.<br/><br/> Though a sync group can have server endpoints on different File Servers, the files cannot be distinct. | *Follow guidance in Scenario # 1 above with additional considerati ...
```
```suggestion
| 5 | Multiple file servers with single (root volume or share) to same target Azure file share (consolidation) | No | A sync group can't use cloud endpoint (Azure file share) already configured in another sync group.<br/><br/> Although a sync group can have server endpoints on different file servers, the files can't be distinct. | *Follow guidance in Scenario # 1 above with additional consideration of targeting one file server at a time.* |
```

jithubhaijabs committed:
Update articles/storage/file-sync/file-sync-planning.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jithubhaijabs committed:
Update includes/storage-files-migration-namespace-mapping.md

Co-authored-by: Ken Downie <60940530+khdownie@users.noreply.github.com>

jborsecnik committed:
Update storage-files-migration-namespace-mapping.md

jborsecnik committed:
Update storage-files-migration-namespace-mapping.md
